cyclic_boosting/generic_loss.py

```diff
@@ -65,31 +65,51 @@ def calc_parameters(
         float, float
             estimated parameters and its uncertainties
         """
-        sorting = feature.lex_binned_data.argsort()
-        sorted_bins = feature.lex_binned_data[sorting]
-        bins, split_indices = np.unique(sorted_bins, return_index=True)
-        split_indices = split_indices[1:]
+        # ! TODO: [Q] Why these operations?
+        #  I probably do not understand the CB algo, high level explainer would be great
```

Comment:
The difference to the other (older) CB modes is that we do the optimizations here numerically by explicitly minimizing a loss, rather than analytically. But in principle, this is the same thing.